docs/advanced/classes.rst

```diff
@@ -159,7 +159,7 @@ Here is an example:
 
```

Comment:
@rwgk, @bstaletic, I was still wondering if the fragment before needs to be adapted: https://github.com/pybind/pybind11/pull/2429/files#diff-47ee721743764a0358552e687404eb8eR145-R150

It's not entirely right (because it doesn't happen anymore), but it's also not wrong. Is it good enough to keep like this?

Reply:
Good point, I agree, changing that sentence in this PR would be best.
Trying:

Otherwise a TypeError is raised for safety, to not leave the memory for the C++ portion of the instance uninitialized, which would generally cause undefined behavior if the instance is used subsequently.

Reply:
My main problem was not how to formulate it, but how to best structure it with this `.. versionchanged:: 2.6` line, actually?

Reply:
Hm, seems fine to me, maybe with tiny tweaks?

> The default pybind11 metaclass was changed to throw a `TypeError` when it detects that `__init__` was not called by a derived class (PR #2152), while prior versions left the C++ portion of the instance uninitialized.

(I'd include the PR number for easy reference for those who know what to do with it; it's easily ignored for those who don't.)

Reply:
> Otherwise a TypeError is raised for safety, to not leave the memory for the C++ portion of the instance uninitialized, which would generally cause undefined behavior if the instance is used subsequently.

Actually, reading this back and trying to fit in these changes, I think it's fine the way it's formulated now. I'm now realizing it's still important to state the whole thing about UB, because it's still the underlying problem (and is still crucial when a non-default metaclass is used). The only difference is that pybind11 now catches this in the default metaclass. So the structure seems fine to me:

> - Call subclass `__init__`
> - Because you need to initialize the underlying class.
> - But the default metaclass will help you (from 2.6 onwards).

Sorry for the confusion!

> (I'd include the PR number for easy reference for those who know what to do with it; it's easily ignored for those who don't.)

@henryiii already included this in the changelog! :-) I'm assuming that people that know what to do with a pybind11 PR would also find the changelog? (I also don't see the convention elsewhere in the docs, so it might be nice to constrain the development reference to the reflog.)